components/dropdown/dropdown-opener-mixin.js

```diff
@@ -188,49 +185,50 @@ export const DropdownOpenerMixin = superclass => class extends superclass {
 		this._isOpenedViaClick = false;
 	}
 
-	/* used by open-on-hover option */
-	__onDropdownMouseEnter() {
-		this._isOpen = true;
-		this._isFading = false;
-		this._closeTimerStop();
-	}
-
-	/* used by open-on-hover option */
-	__onDropdownMouseLeave(e) {
-		if (this.__getContentElement() !== e.target) return;
-		if (!this._isOpenedViaClick) this._isOpen = false;
-		this._closeTimerStart();
-	}
-
 	__onDropdownMouseUp() {
 		this._isOpen = true;
 		this._isFading = false;
+		this._isHovering = false;
 		this._isOpenedViaClick = true;
 		this._closeTimerStop();
 	}
 
 	__onKeypress(e) {
 		if (e.srcElement === this || isComposedAncestor(this.getOpenerElement(), e.srcElement)) {
-			this.__onOpenerKeyPress(e);
+			if (e.keyCode !== 13 && e.keyCode !== 32) return;
```

Comment:
Moved the previous _onOpenerKeypress into this function to get rid of an unneeded extra definition there.